src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.JS.Owned.cs

```diff
@@ -81,6 +84,7 @@ public static void GetTaskSourceTaskRef(int tcsGCHandle, out object result)
         {
             GCHandle handle = (GCHandle)(IntPtr)tcsGCHandle;
             // this is JS owned Normal handle. We always have a Target
+            Debug.Assert(handle.Target != null, "TaskCompletionSource should not be null.");
```

Comment:
Will this assert run (and throw) in Release builds? If not we'll get a NullReferenceException at the end, so we should just turn it into an if -> throw.

Reply:
Sorry, this shouldn't be here. Thanks for the catch.